Mobile/Android/MainPage/BackPressImpl.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Android.Support.V4.App;
+
+namespace Android
+{
+
+	public class BackPressImpl : IBackPressed {
+
+		private Fragment _parentFragment;
+
+		public BackPressImpl(Fragment parentFragment) {
+			
+			_parentFragment = parentFragment;
+		}
+
+		public bool OnBackPressed() {
+
+			if (_parentFragment == null) return false;
+
+			int childCount = _parentFragment.ChildFragmentManager.BackStackEntryCount;
+
+			if (childCount == 0) {
+				return false;
+
+			} else 
```

Comment:
Why do we need the else statement ?. If the condition is true it will return so the code below will never be reach